packages/accounts-base/accounts_server.js

```diff
@@ -1132,21 +1132,52 @@ Ap._expirePasswordResetTokens = function (oldestValidDate, userId) {
 
   oldestValidDate = oldestValidDate ||
     (new Date(new Date() - tokenLifetimeMs));
+
+  var tokenFilter = {
+    $or: [
+      { "services.password.reset.reason": "reset"},
+      { "services.password.reset.reason": {$exists: false}}
+    ]
+  };
+
+  this._expirePasswordToken(oldestValidDate, tokenFilter, userId);
+}
+
+// Deletes expired password enroll tokens from the database.
+//
+// Exported for tests. Also, the arguments are only used by
+// tests. oldestValidDate is simulate expiring tokens without waiting
+// for them to actually expire. userId is used by tests to only expire
+// tokens for the test user.
+Ap._expirePasswordEnrollTokens = function (oldestValidDate, userId) {
+  var tokenLifetimeMs = this._getPasswordResetTokenLifetimeMs("enroll");
+
+  // when calling from a test with extra arguments, you must specify both!
+  if ((oldestValidDate && !userId) || (!oldestValidDate && userId)) {
+    throw new Error("Bad test. Must specify both oldestValidDate and userId.");
+  }
+
+  oldestValidDate = oldestValidDate ||
+    (new Date(new Date() - tokenLifetimeMs));
+
+  var tokenFilter = {
+    "services.password.reset.reason": "enroll"
+  }
+
+  this._expirePasswordToken(oldestValidDate, tokenFilter, userId);
+}
+
+Ap._expirePasswordToken = function (oldestValidDate, tokenFilter, userId) {
```

Comment:
Also, I recommend moving this method above where it's first used (i.e. placed at ~1077)